### ISSUE TITLE
Not an issue but a feature request: batch processing.

### ISSUE THREAD

**Issue by gorman42** [sc0ty/subsync#5]
Hi! I've just tried your awesome program for the first time and... WOW!!!

Do you think it would be possible to add batch processing? Say a directory with video files with correspondingly named subtitles files, all files with same languages involved... press one button and leave the program working in the background?

A side question: is your project somehow related to this: https://github.com/smacke/subsync ?

**Comment by sc0ty** [comment]
Batch processing would be nice but I'm not sure how GUI for it should look like. I was thinking about some kind of table where in each row you could configure things like track numbers, languages etc. and let it select sane defaults automatically (similar to what is already done when you open file). This would require refactoring existing code, since GUI is currently heavily conjoined with application logic (I suck at GUI programming).
Other option would be to have better command line arguments, so you could easily run it in loop from shell or script. But not everybody know how to use shell.

And it is not related to this smacke project, but thanks for the link, it looks interesting. I was thinking of adding speech detection library for performance reasons, and doing speech recognition only when it detect speech. Also like his FFT approach, maybe I could adapt something similar.

**Comment by gorman42** [comment]
Hi! Thanks for your reply. I take the liberty to use this "issue" as a contact to discuss the project. If you object to this, use, feel free to ignore my ramblings. :)

As far as GUI is concerned, you're right... some thought needs to go into this.

What I can tell you after having used the tool is that one of the most likely scenario for batch usage is one where you need to process several similar subtitles. Like many episodes of the same TV series, for instance.
In such a use case, the tool could "simply" pair video+subs with the same name (but different extension) and process them all. Ideally asking only once for the language of the subs and/or video.

I also have a file that yesterday refused to sync correctly, there was no way I could make it work for the first... 6 or 7 minutes of the video. The rest of the video/subs got synced correctly but the first segment needed to be done by hand. If you are interested in examining this, send me a private message (and now that I check there's no private messaging on github... you can write an email to registrazioni.gorman (AT) gmail dot com).

**Comment by sc0ty** [comment]
I know the pain of synchronizing whole season, I've been there myself :)

I was thinking about something similar but maybe with option to select subtitles manually if names don't match. And maybe with global language, encoding, track number, etc. switches that will set this settings for every file, leaving you the option to do it manually. I'd like to have option to configure things per file so you could use this feature for different scenarios, but ideally without zillion clicks for typical/simple use cases.

Feel free to discuss it here further. I think we could work something out.

Another question is when I will finally find some time to implement it, probably not so fast.

**Comment by Kravimir** [comment]
When the file names don't match, you could use a queue setup like Handbrake uses, although that would require not using the UI-blocking "Synchronization" box.

In the mean time, any chance you could change it to remember the last language used for each (audio and text)?

**Comment by sc0ty** [comment]
I've already started implementing something similar: table with two columns for subtitles and references. Each row is single synchronization job.
With drag & drop support.

I'm not sure how to make stream selection in this design. Best what I could think of is window with options like selection by language and type (audio/subtitles/all).

Languages are selected from metadata if available. I could make it to select previously chosen when its not available.

**Comment by logix2** [comment]
How do you use this batch processing on the command line (using ```--cli```)?

**Comment by sc0ty** [comment]
sybsync --cli batch /tasks/list.yaml
And in yaml file:
```
- sub:
    path: '/input/sub1.srt'
    lang: eng
    stream: 1
  ref:
    path: '/input/ref1.avi'
    stream: 2
    lang: eng
  out:
    path: '/out/out1.srt'
- sub:
    path: '/input/sub2.srt'
    lang: eng
    stream: 1
  ref:
    path: '/input/ref2.avi'
    stream: 2
    lang: eng
  out:
    path: '/out/out2.srt'
```
You could generate yaml for your files with gui, in batch sync menu there is export option.
I will describe this and other command line options details in the wiki soon.

**Comment by logix2** [comment]
Is it any way of the cli auto detecting the subtitle/audio stream? Having to specify it for each file with multiple streams kind of defeats the purpose of batch processing...

**Comment by sc0ty** [comment]
stream attribute is optional, without it it will try to select first subtitle/audio stream

**Comment by sc0ty** [comment]
See https://github.com/sc0ty/subsync/wiki/Command-line-options#batch-command

**Comment by logix2** [comment]
Thanks!
